jobs: TestRegistryLifecycle failed

jobs.TestRegistryLifecycle [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3491393&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3491393&tab=artifacts#/) on master @ [85b1e92455eb2795043d71e5766b1b56dab308f2](https://github.com/cockroachdb/cockroach/commits/85b1e92455eb2795043d71e5766b1b56dab308f2):


```
=== RUN   TestRegistryLifecycle
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRegistryLifecycle188845550
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestRegistryLifecycle
    jobs_test.go:1188: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRegistryLifecycle188845550
--- FAIL: TestRegistryLifecycle (7.11s)
=== RUN   TestRegistryLifecycle/OnPauseRequest
    jobs_test.go:254: Starting resume
    jobs_test.go:937: expected job status: paused but got: pause-requested
    jobs_test.go:267: Exiting resume
    jobs_test.go:314: Starting success
    jobs_test.go:318: Exiting success
    --- FAIL: TestRegistryLifecycle/OnPauseRequest (2.60s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestRegistryLifecycle PKG=./pkg/jobs TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-schema spaskob
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegistryLifecycle.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/jobs/BUILD.bazel
+++ b/pkg/jobs/BUILD.bazel
@@ -123,7 +123,6 @@ go_test(
         "//pkg/util/metric",
         "//pkg/util/protoutil",
         "//pkg/util/randutil",
-        "//pkg/util/retry",
         "//pkg/util/stop",
         "//pkg/util/syncutil",
         "//pkg/util/timeutil",

--- a/pkg/jobs/jobs_test.go
+++ b/pkg/jobs/jobs_test.go
@@ -51,7 +51,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
-	"github.com/cockroachdb/cockroach/pkg/util/retry"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
@@ -341,12 +340,7 @@ func (rts *registryTestSuite) tearDown() {
 
 func (rts *registryTestSuite) check(t *testing.T, expectedStatus jobs.Status) {
 	t.Helper()
-	opts := retry.Options{
-		InitialBackoff: 5 * time.Millisecond,
-		MaxBackoff:     time.Second,
-		Multiplier:     2,
-	}
-	if err := retry.WithMaxAttempts(rts.ctx, opts, 10, func() error {
+	testutils.SucceedsSoon(t, func() error {
 		rts.mu.Lock()
 		defer rts.mu.Unlock()
 		if diff := cmp.Diff(rts.mu.e, rts.mu.a); diff != "" {
@@ -363,9 +357,7 @@ func (rts *registryTestSuite) check(t *testing.T, expectedStatus jobs.Status) {
 			return errors.Errorf("expected job status: %s but got: %s", expectedStatus, st)
 		}
 		return nil
-	}); err != nil {
-		t.Fatal(err)
-	}
+	})
 }
 
 func TestRegistryLifecycle(t *testing.T) {

